homeassistant/components/emoncms/sensor.py

```diff
@@ -192,8 +192,10 @@ def __init__(
             self._state = self._value_template.render_with_possible_json_value(
                 elem["value"], STATE_UNKNOWN
             )
-        else:
+        elif elem["value"] is not None:
             self._state = round(float(elem["value"]), DECIMALS)
+        else:
+            self._state = 0
```

Comment:
This seems a bug? Shouldn't it return `None` (which ends up being "unknown") when no value is returned?

Reply:
@frenck, this comment is on the basis I understand emoncms but not the structure within HA.

> This seems a bug?

In emoncms? No, a NULL is returned by design if there is no valid value (bit like HA 'unknown').

The check for a template is possibly superfluous as HA will only ever return a numerical value. However, could a template be used to scale or round for instance?

I think there should be a test for `elem["value"]` being a valid numeric value **before** the template test which, if it fails, sets the state as `unknown`.

I do not think the state should ever be arbitrarily set to 0.

Reply:
> In emoncms?

No not in emoncms, a bug in this PR.

> No, a NULL is returned by design if there is no valid value (bit like HA 'unknown').

Exactly, so that is why retuning `0` here is a bug.

> I do not think the state should ever be arbitrarily set to 0.

I agree, which is why I comment, it sets the state to 0 hardcoded. It should never be `0` IMHO.